udgrade

xzackli opened the pull request:
_udgrade from the fortran_

I also moved UNSEEN to a more prominent location, because it's important for this routine and probably elsewhere.

ziotom78 commented:
Thanks @xzackli , this looks fine for me. Feel free to merge and update CHANGELOG.md when you think it's ready!

xzackli commented:
I've added a test for nside_in=nside_out and fixed a bug with that case, so now it's ready. I don't think I have write privileges for this repo to merge!
